### PR TITLE
Add CLUE external library.

### PR DESCRIPTION
**Pull request by rovere**
This addresses the issue [here](https://github.com/cms-sw/cmssw/issues/45039).

### CODE REVIEW

**Comment by fwyzard** [comment]
please test

**Comment by smuzaffar** [comment]
@rovere , can you please rebase your changes on top of latest IB/CMSSW_14_1_X/master branch and add `clue` dependency in `cmssw-tools.spec`.

**Comment by rovere** [comment]
@smuzaffar let me know if that's what you asked for.

File: cmssw-tools.spec [hunk 1]
```diff
@@ -146,6 +146,7 @@ Requires: dasgoclient
 Requires: dablooms
 Requires: zlib
 Requires: rivet
+Requires: clue
```
Comment (fwyzard) [inline cmssw-tools.spec]: I don't remember why `alpaka` is in the Linux-only section, but could you move `clue` to the same spot, right after `alpaka` itself ?

Comment (rovere) [reply cmssw-tools.spec]: @fwyzard you are right, since clue depends on alpaka.
Did it, thanks.

File: clue.spec [hunk 2]
```diff
@@ -0,0 +1,17 @@
+### RPM external clue 1.0.0
+## NOCOMPILER
+
+%define git_commit V_1_0_0
```
Comment (smuzaffar) [inline clue.spec]: @rovere , can you please also change this to
```
%define git_commit V_%(echo %{realversion} | tr '.' '_')
```

this way in future we only need to update the version in the first line :-)

**Comment by smuzaffar** [comment]
> @smuzaffar let me know if that's what you asked for.

looks good

File: clue.spec (hunk 2 above)
Comment (rovere) [reply clue.spec]: Done, thanks!

Comment (fwyzard) [reply clue.spec]: You could also change the release tag in GitHub to `1.0.0` :-)

**Comment by fwyzard** [comment]
please test

**Comment by srimanob** [comment]
Hi @rappoccio @antoniovilela 
Should this external PR also be merged? Thanks.

**Comment by fwyzard** [comment]
Yes, this external is needed to go with https://github.com/cms-sw/cmssw/pull/45178 :

[`RecoLocalCalo/HGCalRecProducers/plugins/BuildFile.xml`, line 49](https://github.com/rovere/cmssw/blob/d996d3df95d583c9abbdf3d8d0f7b32b83bf18f6/RecoLocalCalo/HGCalRecProducers/plugins/BuildFile.xml#L49)
```xml
  <use name="clue"/>
```

**Comment by smuzaffar** [comment]
+externals

@cms-sw/orp-l2 , I am merging this for 14.1.X as it is needed by  which is already merged

**Pull request merged**